File: angular/src/app/contracts/order.ts
import { ShoppingItem } from './shoppingItem'

export class Order {
  id: string
  date: Date
  total: number
  shoppingItems: ShoppingItem[]

  constructor() {
    this.id = null
    this.date = null
    this.total = null
    this.shoppingItems = []
  }
}

File: aspnet-core/Klir.TechChallenge.Web.Api/Adapter/In/Controllers/RequestsResponses/Order.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;

namespace Klir.TechChallenge.Web.Api.Adapter.In.Controllers.RequestsResponses
{
    [ExcludeFromCodeCoverage]
    public class Order
    {
        public string Id { get; set; }
        public DateTime Date { get; set; }
        public decimal Total { get; set; }
        public IEnumerable<ShoppingItem> ShoppingItems { get; set; }
    }
}